channels/api.py

```diff
@@ -350,6 +354,65 @@ def sync_comment_model(*, channel_name, post_id, comment_id, parent_id):
         )[0]
 
 
+def sync_channel_subscription_model(channel_name, user):
+    """
+    Create or update channel subscription for a user
+
+    Args:
+        channel_name (str): The name of the channel
+        user (django.contrib.models.auth.User): The user
+
+    Returns:
+        ChannelSubscription: the channel user role object
+    """
+    with transaction.atomic():
+        channel = sync_channel_model(channel_name)
+        return ChannelSubscription.objects.update_or_create(channel=channel, user=user)[
+            0
+        ]
+
+
+def get_role_model(channel_name, role):
+    """
+    Get or create a ChannelGroupRole object
+
+    Args:
+        channel_name(str): The channel name
+
+    Returns:
+        ChannelGroupRole: the ChannelGroupRole object
+    """
+    return ChannelGroupRole.objects.get_or_create(
+        channel=Channel.objects.get_or_create(name=channel_name)[0],
+        group=Group.objects.get_or_create(name="{}_{}".format(channel_name, role))[0],
+        role=role,
+    )[0]
+
+
+def add_user_role(channel_name, role, user):
+    """
+    Add a user to a channel role's group
+
+    Args:
+        channel_name(str): The channel name
+        user(django.contrib.auth.models.User): The user
+        role(str): The role name (moderators, contributors)
+    """
+    get_role_model(channel_name, role).group.user_set.add(user)
+
+
+def remove_user_role(channel_name, role, user):
+    """
+    Remove a user from a channel role's group
+
+    Args:
```

Comment:
Could you fix either the argument order or the order they appear in the docstring? Same for `add_user_role`.

Reply:
✅